utils/keys.go

```diff
@@ -103,11 +105,73 @@ func ImportKeys(from io.Reader, to []Importer) error {
 		toWrite []byte
 	)
 	for block, rest := pem.Decode(data); block != nil; block, rest = pem.Decode(rest) {
+		// if there is a path then we set the gun header from this path
+		if rawPath := block.Headers["path"]; rawPath != "" {
+			pathWOFileName := strings.TrimSuffix(rawPath, filepath.Base(rawPath))
+			if strings.HasPrefix(pathWOFileName, notary.NonRootKeysSubdir) {
+				gunName := strings.TrimPrefix(pathWOFileName, notary.NonRootKeysSubdir)
+				block.Headers["gun"] = gunName[1:(len(gunName) - 1)] //removes the slashes
+			}
+		}
+		if block.Headers["gun"] == "" {
+			if fallbackGun != "" {
+				block.Headers["gun"] = fallbackGun
+			}
+		}
+		if block.Headers["role"] == "" {
+			if fallbackRole == "" {
+				block.Headers["role"] = notary.DefaultImportRole
+			} else {
+				block.Headers["role"] = fallbackRole
+			}
+		}
 		loc, ok := block.Headers["path"]
+		// only if the path isn't specified do we get into this parsing path logic
 		if !ok || loc == "" {
-			logrus.Info("failed to import key to store: PEM headers did not contain import path")
-			continue // don't know where to copy this key. Skip it.
+			// if the path isn't specified, we will try to infer the path rel to trust dir from the role (and then gun)
+			// parse key for the keyID which we will save it by.
+			// if the key is encrypted at this point, we will generate an error and continue since we don't know the ID to save it by
+			decodedKey, err := utils.ParsePEMPrivateKey(pem.EncodeToMemory(block), "")
+			if err != nil {
+				logrus.Info("failed to import key to store: Invalid key generated, key may be encrypted and does not contain path header")
+				continue
+			}
+			keyID := decodedKey.ID()
+			if block.Headers["role"] == tufdata.CanonicalRootRole {
```

Comment:
Totally non-blocking nitpick: this might be a good place to use switch, since it shortens everything:

```
switch block.Headers["role"] {
case tufdata.CanonicalRootRole:
    ...
case tufdata.CanonicalSnapshotRole, tufdata.CanonicalTargetsRole, tufdata.CanonicalTimestampRole:
    ...
default:  // delegation
    ...
}
```